web/frontend/src/components/utils/useConfiguration.tsx

```diff
@@ -0,0 +1,165 @@
+import { renderRank, renderSelect, renderSubject, renderText } from 'pages/ballot/QuestionDisplay';
+import { useEffect, useState } from 'react';
+import {
+  Answers,
+  Configuration,
+  Error,
+  ID,
+  Question,
+  RANK,
+  ROOT_ID,
+  RankAnswer,
+  SELECT,
+  SUBJECT,
+  SelectAnswer,
+  Subject,
+  TEXT,
+  TextAnswer,
+} from 'types/configuration';
+
+// Flattens and re-orders the questions (as the first object encountered in a
+// Subject might not be the first that needs to be displayed according to the Order).
+// Adds a ParentID for later rendering and initializes the Answers.
+function flattenSubject(
+  subject: Subject,
+  parentId: ID,
+  sortedQuestions: Array<Question>,
+  outerOrder: number,
+  answerList: Answers
+) {
+  sortedQuestions.push({
+    Order: outerOrder,
+    ParentID: parentId,
+    Type: SUBJECT,
+    Content: subject,
+    render: renderSubject,
+  });
+
+  // The number of Question inside of a top-level Subjects (i.e a with ParentID = ROOT_ID)
+  let numberOfQuestion = 1;
+
+  if (subject.Subjects) {
```

Comment:
In the interface we created, each of the element can be undefined, so it would be necessary. But then again it might be 
better to have them as either null | Subject (text, rank, select ....) ?